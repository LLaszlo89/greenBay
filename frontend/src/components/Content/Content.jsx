import React, { useEffect } from "react";
import ItemCard from "../ItemCard/ItemCard";
import { Grid } from "@material-ui/core";
import { downloadItemsToStore } from "../../redux/actions/itemsActions";
import { connect } from "react-redux";

const Content = ({ download_items, list }) => {

  useEffect(() => {
    download_items();
  }, [download_items]);
  return (
    <>
      {list.length > 0 && (
        <Grid container >
          {list.map((item) => (
            <Grid key={item.id} item xs={12} sm={4}>
              <ItemCard {...item} />
            </Grid>
          ))}
        </Grid>
      )}
    </>
  );
};

const mapStateToProps = (state) => {
  return {
    list: state.items.items,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    download_items: () => dispatch(downloadItemsToStore()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Content);
