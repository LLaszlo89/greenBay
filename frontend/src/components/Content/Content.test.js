import React from "react";
import renderer from "react-test-renderer";
import Content from "./Content";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

describe("Connect component to redux store ", () => {
  const mockStore = configureStore([]);

  const items = {
    items: [
      {
        id: 1,
        user_name: "Test",
        title: "Test",
        description: "Test",
        URL: "Test",
        price: 1,
        forSale: 1,
      },
    ],
  };

  const store = mockStore({ items });

  it("Should render correctly", () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Content />
      </Provider>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
