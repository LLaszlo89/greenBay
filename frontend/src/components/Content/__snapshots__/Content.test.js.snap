// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connect component to redux store  Should render correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
    >
      <div
        className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
            >
              Test
            </span>
            <span
              className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
            >
              1 green_$
            </span>
          </div>
        </div>
        <div
          className="MuiCardMedia-root makeStyles-img-2"
          style={
            Object {
              "backgroundImage": "url(\\"Test\\")",
            }
          }
        />
        <div
          className="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-3 MuiButton-textSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              BUY NOW
            </span>
          </button>
        </div>
        <div
          className="MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Test
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
